src/targets/gpu/kernels/include/migraphx/kernels/softmax.hpp

```diff
@@ -43,7 +44,7 @@ __device__ void softmax(Input input1, Output output)
         auto exp_in = r.inner([&](auto x) { return migraphx::exp(x - c); })(input);
         auto batch_sum =
             r.reduce(op::sum{}, 0, [](auto x) { return migraphx::convert<float>(x); })(exp_in);
-        r.inner([&](auto& y, auto x) { y = x / batch_sum; })(output, exp_in);
+        r.inner([&](auto& y, auto x) { y = static_cast<otype>(x / batch_sum); })(output, exp_in);
```

Comment:
`otype{x / batch_sum}`.

Reply:
Actually, thinking about it more, this should just call the `implicit_conversion` function. We probably need to update it to use the explicit constructor.

Reply:
changed it to `otype{x/batch_sum}`. 

`implict_conversion` is the reason i had to add static_cast in many places. It can't call to explict constructors.

Reply:
I need to use static_cast otherwise it complains about narrowing conversion for __Float16. 

```
In file included from /tmp/comgr-491967/input/main.cpp:3:
/tmp/comgr-491967/include/migraphx/kernels/softmax.hpp:47:50: error: non-constant-expression cannot be narrowed from type 'type' (aka 'float') to 'otype' (aka '_Float16') in initializer list [-Wc++11-narrowing]
        r.inner([&](auto& y, auto x) { y = otype{x / batch_sum}; })(output, exp_in);
```

Reply:
I mean the `implicit_conversion` function [here](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/src/targets/gpu/kernels/include/migraphx/kernels/vec.hpp#L215). You would need to update it [here](https://github.com/ROCmSoftwarePlatform/AMDMIGraphX/blob/develop/src/targets/gpu/kernels/include/migraphx/kernels/vec.hpp#L210) to use `U(x)` instead of `{x}`.

Reply:
I tried that with `U{x}` 
but for the _FLoat16 it requires explicit cast still. 

tmp/comgr-55f84c/include/migraphx/kernels/vec.hpp:210:18: error: non-constant-expression cannot be narrowed from type 'float' to '_Float16' in initializer list [-Wc++11-narrowing]
        return U{x};
                 ^